### PR TITLE
feat: add new tab to migrate helm release in CD

### PR DESCRIPTION
**Pull request by arunjaindev**
# Description
Add support for migrating helm release in CD Build Pipeline

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/css/base.scss [hunk 1]
```diff
@@ -2191,6 +2191,10 @@ button.anchor {
     border: 1px solid var(--N700);
 }
 
+.dc__border-b2 {
```
Comment (AbhishekA1509) [inline src/css/base.scss:2194]: can use eb-2 bw-1, please revert

File: src/components/cdPipeline/BuildCD.tsx [hunk 2]
```diff
@@ -359,26 +364,15 @@ export default function BuildCD({
         setFormData(_form)
     }
 
-    const renderEnvNamespaceAndTriggerType = () => {
+    const  renderEnvSelector = () => {
```
Comment (AbhishekA1509) [inline src/components/cdPipeline/BuildCD.tsx:367]: please format

File: src/components/cdPipeline/BuildCD.tsx [hunk 3]
```diff
@@ -392,6 +386,66 @@ export default function BuildCD({
...
+        return (
+            <>
+                <div className="form__label dc__required-field">Environment</div>
+                <ReactSelect
+                    menuPosition={isAdvanced ? null : 'fixed'}
+                    closeMenuOnScroll
+                    isDisabled={!!cdPipelineId}
+                    classNamePrefix="cd-pipeline-environment-dropdown"
+                    placeholder="Select Environment"
+                    options={releaseMode === ReleaseMode.LINK ? envList.filter((env) => !env.isVirtualEnvironment) : envList}
```
Comment (AbhishekA1509) [inline src/components/cdPipeline/BuildCD.tsx:399]: Can we rename ReleaseMode.Link to something readable

File: src/components/cdPipeline/BuildCD.tsx [hunk 4]
```diff
@@ -392,6 +386,66 @@ export default function BuildCD({
...
+            </>
+        )
+    }
+
+    const renderEnvNamespaceAndTriggerType = () => {
+        const envId = formData.environmentId
+        const selectedEnv: Environment = formData.environments.find((env) => env.id == envId)
+        const namespaceEditable = false
+
+        const getNamespaceplaceholder = (): string => {
```
Comment (AbhishekA1509) [inline src/components/cdPipeline/BuildCD.tsx:439]: Can follow camelCase

File: src/components/cdPipeline/BuildCD.tsx [hunk 5]
```diff
@@ -392,6 +386,66 @@ export default function BuildCD({
...
+            if (isVirtualEnvironment) {
```
Comment (AbhishekA1509) [inline src/components/cdPipeline/BuildCD.tsx:440]: Can we move this method outside this method it is anti pattern to have methods in methods

File: src/components/cdPipeline/CDPipeline.tsx [hunk 6]
```diff
@@ -1194,6 +1212,13 @@ export default function CDPipeline({
             title = CREATE_DEPLOYMENT_PIPELINE
         }
 
+        // Disable button if environment or release name is not selected
+        const getButtonDisabledMessage = (): string => {
```
Comment (AbhishekA1509) [inline src/components/cdPipeline/CDPipeline.tsx:1216]: please wrap in curly braces

**Review by AbhishekA1509 — CHANGES_REQUESTED** [review]

**Review by AbhishekA1509 — DISMISSED** [review]

File: src/components/cdPipeline/cdpipeline.util.tsx [hunk 7]
```diff
@@ -425,3 +426,13 @@ export const filterInvalidConditionDetails = (
         }) || []
     )
 }
+
+export const getNamespacePlaceholder = (isVirtualEnvironment: boolean, namespace: string): string => {
+    if (isVirtualEnvironment) {
+        if (namespace) {
```
Comment (vivek-devtron) [inline src/components/cdPipeline/cdpipeline.util.tsx:432]: We should reverse this check to remove duplicity

File: src/components/app/details/appDetails/SourceInfo.tsx [hunk 8]
```diff
@@ -277,22 +279,22 @@ export const SourceInfo = ({
                               />
                           )}
                           {isVirtualEnvironment && renderGeneratedManifestDownloadCard()}
-                          {!loadingResourceTree && (
+                          {!loadingResourceTree && !helmMigratedAppNotTriggered && (
```
Comment (vivek-devtron) [inline src/components/app/details/appDetails/SourceInfo.tsx:282]: We should add wrap a common check of `!helmMigratedAppNotTriggered`

File: src/components/cdPipeline/BuildCD.tsx [hunk 9]
```diff
@@ -637,7 +644,7 @@ export default function BuildCD({
             <>
                 <p className="fs-14 fw-6 cn-9 mb-8 mt-16">Deployment Strategy</p>
                 <p className="fs-13 fw-5 cn-7 mb-8">Configure deployment preferences for this pipeline</p>
-                <ReactSelect
+                <ReactSelect<OptionType, false>
```
Comment (vivek-devtron) [inline src/components/cdPipeline/BuildCD.tsx:647]: can we use common selectPicker here?

**Review by vivek-devtron — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**